krun/platform.py

```diff
@@ -761,23 +762,61 @@ def find_temperature_sensors(self):
         The sensors move between reboots, so we have to roll our own sensor
         identifier. We are not fussy, and use as many sensors as we can find."""
 
+        ignore_unnamed_chips = False
         # maps our identifier to actual sysfs filename
         sensor_map = {}
+        # Keeps track of which chip names we have already seen, so that we can
+        # ensure each chip has a unique name by which we can identify it.
+        chips_seen = []
 
         hwmons = glob.glob(LinuxPlatform.HWMON_CHIPS_GLOB)
         for chip in hwmons:
             name_path = os.path.join(chip, "name")
-            with open(name_path) as fh:
-                chip_name = fh.read().strip()
+            try:
+                with open(name_path) as fh:
+                    chip_name = fh.read().strip()
+            except IOError:
+                # The chip does not have a name.
+                # That's fine as long as there is only one chip like this.
+                debug("Un-named temperature sensor chip found.")
+                chip_name = LinuxPlatform.UNKNOWN_SENSOR_CHIP_NAME
+
+            if chip_name in chips_seen:
```

Comment:
Shouldn't this `if` clause be in the `except` clause?

Reply:
I want to know if there is a duplicate chip name, whether it is "unknown" or otherwise.

Reply:
OK, I get it. I'm not totally keen on the `assert chip_name == LinuxPlatform.UNKNOWN_SENSOR_CHIP_NAME` though, because given all the weirdness we've seen, it seems likely that some poor soul will have duplicates that aren't unnamed.  Maybe we should have something like:

```python
if chip_name != LinuxPlatform.UNKNOWN_SENSOR_CHIP_NAME:
    fatal("Duplicated chip named '%s'")
```
so that it's a bit clearer what happened?